### PR TITLE
[fix](multi-catalog)fix minio default region and throw minio error msg, support s3 bucket root path

### PR DESCRIPTION
**Pull request by wsjz**
## Proposed changes

1. check minio region, set default region if user region is not provided, and throw minio error msg
2. support read root path s3://bucket1
3. fix max compute public access

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by wsjz** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/datasource/property/PropertyConverter.java [hunk 1]
```diff
@@ -309,8 +309,10 @@ private static Map<String, String> convertToCOSProperties(Map<String, String> pr
     }
 
     private static Map<String, String> convertToMinioProperties(Map<String, String> props, CloudCredential credential) {
-        // minio does not have region, use an arbitrary one.
-        props.put(MinioProperties.REGION, "us-east-1");
+        if (!props.containsKey(MinioProperties.REGION)) {
+            LOG.warn("Region has not set, use default s3 region: 'us-east-1'.");
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/property/PropertyConverter.java:313]:
```suggestion
            LOG.debug("Region has not set, use default s3 region: 'us-east-1'.");
```

File: fe/fe-core/src/main/java/org/apache/doris/datasource/property/S3ClientBEProperties.java [hunk 2]
```diff
@@ -23,20 +23,25 @@
...
     /**
      *  convert FE properties to BE S3 client properties
      *  On BE, should use properties like AWS_XXX.
      */
     public static Map<String, String> getBeFSProperties(Map<String, String> properties) {
         if (properties.containsKey(MinioProperties.ENDPOINT)) {
-            // minio does not have region, use an arbitrary one.
-            properties.put(MinioProperties.REGION, "us-east-1");
+            if (!properties.containsKey(MinioProperties.REGION)) {
+                LOG.warn("Region has not set, use default s3 region: 'us-east-1'.");
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/property/S3ClientBEProperties.java:42]:
```suggestion
                LOG.debug("Region has not set, use default s3 region: 'us-east-1'.");
```

File: fe/fe-core/src/main/java/org/apache/doris/tablefunction/S3TableValuedFunction.java [hunk 3]
```diff
@@ -77,8 +78,13 @@ public S3TableValuedFunction(Map<String, String> params) throws AnalysisExceptio
         final String endpoint = forceVirtualHosted
                 ? getEndpointAndSetVirtualBucket(params)
                 : s3uri.getBucketScheme();
+        if (!tvfParams.containsKey(S3Properties.REGION)) {
+            String region = S3Properties.getRegionOfEndpoint(endpoint);
+            LOG.warn("Region has not set, use default s3 region: {}", region);
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/tablefunction/S3TableValuedFunction.java:83]:
```suggestion
            LOG.debug("Region has not set, use default s3 region: {}", region);
```

**Review by morningman — DISMISSED** [review]
LGTM

**Comment by morningman** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/datasource/property/PropertyConverter.java [hunk 4]
```diff
@@ -309,8 +309,9 @@ private static Map<String, String> convertToCOSProperties(Map<String, String> pr
     }
 
     private static Map<String, String> convertToMinioProperties(Map<String, String> props, CloudCredential credential) {
-        // minio does not have region, use an arbitrary one.
-        props.put(MinioProperties.REGION, "us-east-1");
+        if (!props.containsKey(MinioProperties.REGION)) {
+            props.put(MinioProperties.REGION, "us-east-1");
```
Comment (zddr) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/property/PropertyConverter.java:313]: if need a constant for “us-east-1”

File: fe/fe-core/src/main/java/org/apache/doris/datasource/property/S3ClientBEProperties.java [hunk 5]
```diff
@@ -23,20 +23,24 @@
...
     /**
      *  convert FE properties to BE S3 client properties
      *  On BE, should use properties like AWS_XXX.
      */
     public static Map<String, String> getBeFSProperties(Map<String, String> properties) {
         if (properties.containsKey(MinioProperties.ENDPOINT)) {
-            // minio does not have region, use an arbitrary one.
-            properties.put(MinioProperties.REGION, "us-east-1");
+            if (!properties.containsKey(MinioProperties.REGION)) {
+                properties.put(MinioProperties.REGION, "us-east-1");
```
Comment (zddr) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/property/S3ClientBEProperties.java:42]: same

**Comment by wsjz** [comment]
run buildall

**Comment by wsjz** [comment]
run buildall

**Comment by wsjz** [comment]
run buildall

**Comment by wsjz** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 46.5 seconds
 stream load tsv:          505 seconds loaded 74807831229 Bytes, about 141 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 insert into select:          29.8 seconds inserted 10000000 Rows, about 335K ops/s
 storage size: 17162996857 Bytes

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 44.65 seconds
 stream load tsv:          501 seconds loaded 74807831229 Bytes, about 142 MB/s
 stream load json:         19 seconds loaded 2358488459 Bytes, about 118 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 insert into select:          28.7 seconds inserted 10000000 Rows, about 348K ops/s
 storage size: 17168117075 Bytes

**Comment by wsjz** [comment]
run buildall

**Review by zddr — APPROVED** [review]
LGTM

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**